### PR TITLE
remove-v2-wrappers

### LINKED ISSUE #33: Remove V2 references from internal modules
We simply need to remove any V2 wrapper modules or function names. I rewrote the "core" module and used "V2 wrapper modules" to build the newer functionality while using some of the old. All of the old code should all be deleted or unused (or close to it) and the `V2` modules can be removed and become top level in their files

### PR DESCRIPTION
**Pull request by 2xrJohann**
Resolves #33 

done - 
  remove v2 wrappers around internal modules
  replace copy_file implementation with copy_file_v2 implementation & update calls to new name

### CODE REVIEW

File: src/init/scaffold.re [hunk 1]
```diff
@@ -1000,13 +899,114 @@ module V2 = {
...
+               <Badge color=`green> {React.string("Complete")} </Badge>
+               <Text> {React.string("Building project with Dune")} </Text>
+             </Box>
+           : <Spinner label="Building project with Dune" />}
+      </Box>;
+    };
+  };
+};
 
   module Scaffold = {
```
Comment (dmmulroy) [inline src/init/scaffold.re:1011]: We can also remove this module/wrapper. That'll fix in `component.re` accessing it via `module Scaffold = Scaffold.Scaffold` and will instead be just  `module Scaffold = Scaffold`

File: src/init/component.re [hunk 2]
```diff
@@ -2,7 +2,7 @@
 open Bindings;
 open Ink;
 open Common;
-module Scaffold = Scaffold.V2.Scaffold;
+module Scaffold = Scaffold.Scaffold;
```
Comment (dmmulroy) [inline src/init/component.re:5]: See my comment in `scaffold.re`

**Review by dmmulroy — COMMENTED** [review]
This looks great so far and is exactly what we're aiming for. I super appreciate your time and effort

File: src/init/scaffold.re (hunk 1 above)
Comment (2xrJohann) [reply src/init/scaffold.re:1011]: sure thing, will do and push in a bit

Comment (2xrJohann) [reply src/init/scaffold.re]: should be good now :smile:

**Comment by 2xrJohann** [comment]
@dmmulroy I moved this into ready because I think it covers everything in the issue however the current main branch doesnt build

im happy to wait for that if you like so I can make sure these changes are safe, otherwise review at your own leisure :v:

**Comment by dmmulroy** [comment]
I'm grinding through fixing Vite HMR - should be stable here soon!

**Comment by dmmulroy** [comment]
Thank you for your contribution @2xrJohann!!

**Pull request merged**